packages/opensrp-plans/src/components/ActivateMission/index.tsx

```diff
@@ -0,0 +1,75 @@
+import { PlanDefinition, PlanStatus } from '@opensrp/plan-form-core';
+import { OpenSRPService } from '../../helpers/dataLoaders';
+import React from 'react';
+import { Card, Button, Typography } from 'antd';
+import {
+  ACTIVATE_MISSION,
+  FAILED_TO_ACTIVATE_MISSION,
+  SUCCESSFULLY_ACTIVATED_MISSION,
+} from '../../lang';
+import { CommonProps, postPutPlan } from '@opensrp/plan-form';
+import { defaultCommonProps } from '../../helpers/common';
+import { sendErrorNotification, sendSuccessNotification } from '@opensrp/notifications';
+
+const { Title } = Typography;
+
+/** props for the activate mission component */
+interface ActivateMissionProps extends CommonProps {
+  plan: PlanDefinition | null;
+  serviceClass: typeof OpenSRPService;
+  submitCallback: (plan: PlanDefinition) => void;
+}
+
+const defaultProps = {
+  ...defaultCommonProps,
+  plan: null,
+  serviceClass: OpenSRPService,
+  submitCallback: () => {
+    return;
+  },
+};
+
+/** renders a card with a button to effect the activate mission worfklow
+ *
+ * @param props - the component props
+ */
+const ActivateMissionCard = (props: ActivateMissionProps) => {
+  const { plan, serviceClass, baseURL, submitCallback } = props;
+  const planIsDraft = plan?.status === PlanStatus.DRAFT;
+
+  if (!plan || !planIsDraft) {
+    return null;
```

Comment:
why null?

Reply:
this component is only supposed to be shown when the plan is of draft status, the alternative was to have the parent component render it depending on the same check, I however chose to have that check inside the component itself.